### PR TITLE
Upgrade the version of kind and controller-gen to adapt k8s version upgrade

### PR DESCRIPTION
**Pull request by guoqinwill**
Upgrade the version of kind and controller-gen  to adapt k8s version upgrade.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -108,10 +108,6 @@ manifests: controller-gen
 	$(CONTROLLER_GEN) $(CRD_OPTIONS) paths="./vendor/volcano.sh/apis/pkg/apis/scheduling/v1beta1;./vendor/volcano.sh/apis/pkg/apis/batch/v1alpha1;./vendor/volcano.sh/apis/pkg/apis/bus/v1alpha1;./vendor/volcano.sh/apis/pkg/apis/nodeinfo/v1alpha1" output:crd:artifacts:config=config/crd/volcano/bases
 	# generate volcano job crd yaml without description to avoid yaml size limit when using `kubectl apply`
 	$(CONTROLLER_GEN) $(CRD_OPTIONS_EXCLUDE_DESCRIPTION) paths="./vendor/volcano.sh/apis/pkg/apis/batch/v1alpha1" output:crd:artifacts:config=config/crd/volcano/bases
-	# volcano crd v1beta1
-	$(CONTROLLER_GEN) "crd:crdVersions=v1beta1" paths="./vendor/volcano.sh/apis/pkg/apis/scheduling/v1beta1;./vendor/volcano.sh/apis/pkg/apis/batch/v1alpha1;./vendor/volcano.sh/apis/pkg/apis/bus/v1alpha1;./vendor/volcano.sh/apis/pkg/apis/nodeinfo/v1alpha1" output:crd:artifacts:config=config/crd/volcano/v1beta1
```
Comment (lowang-bh) [inline Makefile:112]: Why those code is removed?

Comment (guoqinwill) [reply Makefile:112]: After updating the controller-gen tool to version 0.14.0 (which is compatible with Kubernetes 1.29), generating v1beta1 CRDs is no longer supported. While existing v1beta1 CRDs can still be used, they will not receive further updates.

File: docs/design/adapt-k8s-todo.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Motivation
+
+In order to ensure that volcano is compatible with the new version of kubernetes, the k8s version of volcano needs to be routinely upgraded.
+
+
+## TODO things for upgrade
+
+####  Upgrade volcano apis
+First,we need to upgrade the k8s version of volcano apis, such as:https://github.com/volcano-sh/apis/pull/124.
```
Comment (Monokaix) [inline docs/design/adapt-k8s-todo.md:11]: Here should list what we need to do when update apis repo.
And also add to add that go version in go mod file and go build version of dockerfile should keep consistent with k8s.

File: hack/generate-yaml.sh [hunk 3]
```diff
@@ -86,22 +86,15 @@ JOBFLOW_CRD_DIR=${VK_ROOT}/config/crd/jobflow
...
-tail -n +3 ${VOLCANO_CRD_DIR}/bases/scheduling.volcano.sh_queues.yaml > ${HELM_VOLCANO_CRD_DIR}/bases/scheduling.volcano.sh_queues.yaml
-tail -n +3 ${VOLCANO_CRD_DIR}/bases/nodeinfo.volcano.sh_numatopologies.yaml > ${HELM_VOLCANO_CRD_DIR}/bases/nodeinfo.volcano.sh_numatopologies.yaml
-
-# sync v1beta1
-tail -n +3 ${VOLCANO_CRD_DIR}/v1beta1/batch.volcano.sh_jobs.yaml > ${HELM_VOLCANO_CRD_DIR}/v1beta1/batch.volcano.sh_jobs.yaml
-tail -n +3 ${VOLCANO_CRD_DIR}/v1beta1/bus.volcano.sh_commands.yaml > ${HELM_VOLCANO_CRD_DIR}/v1beta1/bus.volcano.sh_commands.yaml
-tail -n +3 ${VOLCANO_CRD_DIR}/v1beta1/scheduling.volcano.sh_podgroups.yaml > ${HELM_VOLCANO_CRD_DIR}/v1beta1/scheduling.volcano.sh_podgroups.yaml
-tail -n +3 ${VOLCANO_CRD_DIR}/v1beta1/scheduling.volcano.sh_queues.yaml > ${HELM_VOLCANO_CRD_DIR}/v1beta1/scheduling.volcano.sh_queues.yaml
-tail -n +3 ${VOLCANO_CRD_DIR}/v1beta1/nodeinfo.volcano.sh_numatopologies.yaml > ${HELM_VOLCANO_CRD_DIR}/v1beta1/nodeinfo.volcano.sh_numatopologies.yaml
+tail -n +2 ${VOLCANO_CRD_DIR}/bases/batch.volcano.sh_jobs.yaml > ${HELM_VOLCANO_CRD_DIR}/bases/batch.volcano.sh_jobs.yaml
```
Comment (Monokaix) [inline hack/generate-yaml.sh:89]: Why is this changed?

Comment (guoqinwill) [reply hack/generate-yaml.sh:89]: The YAML generated by controller-gen v0.14.0 only has empty lines at the beginning of the file. Removing the first two lines should be sufficient. In previous versions, it was necessary to remove the first three empty lines.

File: docs/design/adapt-k8s-todo.md (hunk 2 above)
Comment (guoqinwill) [reply docs/design/adapt-k8s-todo.md:11]: done

File: .github/workflows/e2e_parallel_jobs.yaml [hunk 4]
```diff
@@ -31,7 +31,7 @@ jobs:
 
       - name: Install dependences
         run: |
-          GO111MODULE="on" go install sigs.k8s.io/kind@v0.15.0
+          GO111MODULE="on" go install sigs.k8s.io/kind@v0.21.0
           curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.23.0/bin/linux/amd64/kubectl && sudo install kubectl /usr/local/bin/kubectl
```
Comment (googs1025) [inline .github/workflows/e2e_parallel_jobs.yaml:35]: can we also upupgrade kubectl to v1.29.0?
refer to: https://github.com/volcano-sh/volcano/pull/3460#issuecomment-2105589681

Comment (guoqinwill) [reply .github/workflows/e2e_parallel_jobs.yaml]: Yes，I will upgrade it.

**Comment by Monokaix** [comment]
Plz make ci happy.

**Review by william-wang — APPROVED** [review]
/approve
